### ISSUE TITLE
Example for image retrieval 

### ISSUE THREAD

**Issue by Zumbalamambo** [KevinMusgrave/pytorch-metric-learning#274]
How do I train the model for image retrieval? Is there any examples?

**Comment by KevinMusgrave** [comment]
During training, you can feed the output of your model to a metric learning loss.
```python
# during training
from pytorch_metric_learning.losses import ContrastiveLoss
loss_fn = ContrastiveLoss()
embeddings = model(img)
loss = loss_fn(embeddings, labels)
```

During inference, you can retrieve images based on a nearest neighbor search of your database of images.

You might find the [example notebooks](https://github.com/KevinMusgrave/pytorch-metric-learning/tree/master/examples) helpful, especially "MNIST using TripletMarginLoss" and "Inference".
